packages/hasura-storage-js/src/utils/upload.ts

```diff
@@ -87,7 +87,7 @@ export const fetchUpload = async (
     xhr.responseType = 'json'
 
     xhr.onload = () => {
-      if (xhr.status < 200 && xhr.status >= 300) {
+      if (xhr.status < 200 || xhr.status >= 300) {
         return resolve({
           fileMetadata: null,
           error: { error: xhr.statusText, message: xhr.statusText, status: xhr.status }
```

Comment:
The error here is in the response

```suggestion
          error: xhr.response?.error ?? xhr.response,
          message: xhr.response?.error?.message ?? xhr.response,
          status: xhr.status
```

Reply:
Updated https://github.com/nhost/nhost/pull/2190/commits/08fe4cd65f717ace5ff0f6b435a4b33bf37eaf77